### ISSUE TITLE
Which version of cowboy should I use?

### ISSUE THREAD

**Issue by drozzy** [ninenines/cowboy#1146]
Looking at the hex.pm package for cowboy, and the documentation, it is not clear which version of cowboy one should use.

I assume this is version 1:
https://hex.pm/packages/cowboy
(it says 1.1.2)

How would one install version 2? (Since the doc seems to favor that: https://ninenines.eu/docs/en/cowboy/2.0/guide/getting_started/)

**Comment by drozzy** [comment]
Ok, I found one way of installing pre-release. Still, would be nice to have it on hex:
```
{ :cowboy, github: "ninenines/cowboy", tag: "2.0.0-pre.7" }
```

**Comment by drozzy** [comment]
Also, it would be nice if cowboy version 2 was called `cowboy2`. Since the api is different, it is rather confusing, when one follows the doc for one version, but it doesn't work because another version is being used.

**Comment by essen** [comment]
The API being different is why we have versions. I recommend using the 2.0 pre-releases for all new projects.

I am not the maintainer of the hex package so feel free to create a cowboy2 package if you think that's a good idea. I wouldn't know.

**Comment by Gazler** [comment]
@drozzy I am one of the maintainers of the hex package. I would not recommend creating a cowboy2 package, the project version is enough to determine that, and semantic versioning dictates that a major release can contain breaking API changes http://semver.org/

Once there is a stable release of Cowboy 2, the hex package will be pushed. For now, the way you have identified is the best way to install from hex. Please be sure to use Elixir 1.4.2 or above as there was an incompatibility with compiling packages which used erlang.mk (such as the new set versions of ranch, cowlib and cowboy).

There are some examples of using Cowboy2 with Plug and Phoenix linked to on this issue https://github.com/elixir-lang/plug/issues/258#issuecomment-274052599

**Comment by drozzy** [comment]
@Gazler Thank you. I am just pointing out that it is a bit weird that the cowboy site has the docs for Cowboy 2 as the main documentation, but the hex.pm repo only provides Cowboy 1.

I can see many newbies being tripped up by it. Also, it has been in pre-release for a while now, so I'm not sure how much longer it will take. Just wanted to point out this confusion. I know both Erlang and Elixir - and it tripped me up. Spent half an hour trying to understand why my cowboy code wasn't working.

P.S. I'm not using phoenix, personally.

**Comment by essen** [comment]
2.0 will be released this summer.

**Comment by drozzy** [comment]
@essen Thanks. This wasn't to be a critique of your timeline.
@Gazler My point was that some frameworks publish pre-releases to hex, e.g.:
https://hex.pm/packages/phoenix
Has versions like 1.3.0-rc.0

**Comment by essen** [comment]
I know, was just an FYI.

**Comment by drozzy** [comment]
@essen Thank you very much for putting this on hex.pm!

**Comment by essen** [comment]
I didn't. At the moment the Elixir/Phoenix people do it. I'll ask for ownership at a future date once hex_erl is complete.
